src/App.tsx

```diff
@@ -15,36 +17,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+interface SortingParams {
+  sortField: string;
+  isReversed: boolean;
+}
+
+enum SortType {
+  Alphabet = 'Alphabet',
+  Length = 'Length',
+  Default = '',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed }: SortingParams,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.Alphabet:
+          return good1.localeCompare(good2);
+
+        case SortType.Length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
+
+  const resetFilters = () => {
+    setSortField(SortType.Default);
+    setIsReversed(false);
+  };
+
+  const isActive = sortField !== SortType.Default || isReversed;
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortType.Alphabet,
```

Comment:
[UX] - Don't forget to mark the active button to explain to the user what action is active. It seems you've attempted to do so with the 'is-light' class, but ensure it clearly indicates the active state to the user.